Sming/Services/Yeelight/YeelightBulb.cpp

```diff
@@ -157,33 +156,29 @@ bool YeelightBulb::onResponse(TcpClient& client, char* data, int size)
 			p2 = source.length();
 		String buf = source.substring(p, p2);
 		p = unsigned(p2) + 2;
-		DynamicJsonBuffer jsonBuffer;
-		JsonObject& root = jsonBuffer.parseObject(buf);
-		bool parsed = root.success();
-		if (parsed)
+		DynamicJsonDocument doc(1024);
+		auto error = deserializeJson(doc, buf);
+		if (!error)
 		{
-			if (root.containsKey("id") && root.containsKey("result"))
+			long id = doc["id"] | -1;
+			if (id == propsId)
 			{
-				long id = root["id"];
-				if (id == propsId)
-				{
-					auto &result = root["result"].asArray();
-					String resp = result[0].asString();
+				JsonArray result = doc["result"];
+				String resp = result[0] | "";
+				if(resp.length()) {
 					parsePower(resp);
 				}
 			}
```

Comment:
I'd be tempted to rewrite like this:

```
String resp = result[0].as<const char*>;
if(resp) {
	parsePower(resp);
}
```

Because we're testing for the existence of a value, not so much providing a default.